src/python/pants/build_graph/target.py

```diff
@@ -726,16 +682,57 @@ def strict_dependencies(self, dep_context):
     """
     strict_deps = self._cached_strict_dependencies_map.get(dep_context, None)
     if strict_deps is None:
+      closure_kwargs = dep_context.target_closure_kwargs
+      respect_intransitive = closure_kwargs.get('respect_intransitive')
+      include_scopes = closure_kwargs.get('include_scopes')
+      exclude_scopes = closure_kwargs.get('exclude_scopes')
+      def dep_predicate(source, dependency):
+        # Always expand aliases.
+        if type(source) in dep_context.alias_types:
+          return True
+
+        # If the dependency isn't in scope, don't use it.
```

Comment:
The scope and transitivity checks are duplicated with the default predicate... would probably be worthwhile to just call it to avoid them going out of sync.